sdk/core/System.ClientModel/tests/Pipeline/ClientLoggingPolicyTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Options;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using ClientModel.Tests;
+using ClientModel.Tests.Mocks;
+using NUnit.Framework;
+using SyncAsyncTestBase = ClientModel.Tests.SyncAsyncTestBase;
+
+namespace System.ClientModel.Tests.Pipeline;
+
+public class ClientLoggingPolicyTests : SyncAsyncTestBase
+{
+    private const int RequestEvent = 1;
+    private const int ResponseEvent = 5;
+
+    public ClientLoggingPolicyTests(bool isAsync) : base(isAsync)
+    {
+    }
+
+    [Test]
+    public async Task MultiplePipelinesCanLog()
```

Comment:
Just making sure I understand -- is this test testing the case that there are two independent pipelines, each with independent logging policies, but the events from both of them can be observed by a single event listener?

Reply:
That but also checking the singleton logic works when the event source names are the same but there are multiple logging policy instances.